TESTING.md

```diff
@@ -1,89 +1,483 @@
+1. Obtain the IP and port of the Service exposing the UI of the application and adapt `cypress.json` to these values
+
+    ```bash
+    $ kubectl get svc
+    NAME         TYPE           CLUSTER-IP    EXTERNAL-IP     PORT(S)                      AGE
+    kubernetes   ClusterIP      10.99.32.1    <none>          443/TCP                      3m32s
+    nginx        LoadBalancer   10.99.41.53   1.1.1.1   80:31102/TCP,444:30230/TCP   65s
+
+    $ cd .vib/nginx/cypress
+    $ cat cypress.json
+    {
+      "baseUrl": "http://localhost"
+    }
+    # Edit the file to point to 1.1.1.1:80
+    $ nano cypress.json
+    $ cat cypress.json
+    {
+      "baseUrl": "http://1.1.1.1:80"
+    }
+    ```
+1. Launch Cypress indicating the folder where tests are located
+
+    ```bash
+    $ cypress run .
+
+    =====================================================================================
+
+      (Run Starting)
+
+      ┌─────────────────────────────────────────────────────────────────────────────────┐
+      │ Cypress:        9.5.4                                                           │
+      │ Browser:        Electron 94 (headless)                                          │
+      │ Node Version:   v16.13.2 (/usr/local/bin/node)                                  │
+      │ Specs:          1 found (nginx_spec.js)                                         │
+      └─────────────────────────────────────────────────────────────────────────────────┘
+    ...
+    ✔  All specs passed!                        371ms        1        1
+    ```
+
+### Useful information
+
+* In most cases, a single test which covers the following topics is enough:
+  * Login/Logout: Checks the UI, app, and DB are working together
+  * Basic functionality test
+
+* Avoid multiple tests for the same block of functionality.
+
+* If the asset exposes an API, Cypress is an excellent option to test this feature!
+
+### Specific acceptance criteria
+
+* [ ] Test file name has the following format: Helm chart name + spec (ex: `wordpress_spec.js`)
+* [ ] No `describe()` blocks should be present
+* [ ] Aim to have an assertion after every command to avoid flakiness, taking advantage of Cypress retry-ability
+* [ ] Test description is a sentence with the following format: Expected result summary, starting with a verb, in third person, no dots at the end of the sentence (ex: `it('checks if admin can edit a site', ()`)
+* [ ] Respect the folder structure recommended by Cypress:
   * [fixtures](https://docs.cypress.io/api/commands/fixture) - for test data
   * [Integration](https://docs.cypress.io/api/commands/fixture) - test scenario
   * [plugins](https://docs.cypress.io/guides/tooling/plugins-guide) - plugin configuration, if applicable
   * [support](https://docs.cypress.io/api/commands/fixture) - reusable behaviours and overrides
   * [cypress.json](https://docs.cypress.io/guides/tooling/plugins-guide) - configuration values you wish to store
```

Comment:
We should revisit these links and maybe add https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests